eventeditor/actor_json.py

```diff
@@ -0,0 +1,121 @@
+from enum import IntEnum
+import typing
+
+import json
+from pathlib import Path
+import PyQt5.QtWidgets as q
+
+_actor_json_path: typing.Optional[Path] = None
+def set_actor_json_path(p: typing.Optional[str]) -> None:
+    if p:
+        global _actor_json_path
+        _actor_json_path = Path(p)
+
+class EventType(IntEnum):
+    Action = 0
+    Query = 1
+
+def load_actor_json(actor_name: str) -> typing.Dict[str, typing.Any]:
+    if not _actor_json_path:
+        return None
+
+    try:
+        # First look for individual actor file (overrides; for power users)
+        with open(_actor_json_path.parent/f'{actor_name}.json', 'rt') as file:
+            return json.loads(file.read())
+    except:
+        try:
+            # Otherwise look in actor definitions file
+            with open(_actor_json_path, 'rt') as file:
+                return json.loads(file.read())[actor_name]
+        except:
+            return None
+
+def load_event_parameters(actor_name: str, event_name: str, event_type: EventType) -> typing.Dict[str, typing.Any]:
+    try:
+        actor = load_actor_json(actor_name)
+
+        if event_type == EventType.Action:
+            return actor['actions'][event_name]
+        if event_type == EventType.Query:
+            return actor['queries'][event_name]
+        return None
+    except:
+        return None
+
+def load_actions(actor_name: str) -> typing.KeysView[str]:
```

Comment:
`Iterable[str]` (callers don't need to know that it's a KeysView, just that it's something that can be iterated on and that produces strings)